### PR TITLE
parse constants/enums

### PR DESCRIPTION
**Pull request by Gummibeer**
this solves #42

### CODE REVIEW

File: generator/Parser/DefinitionParser.php [hunk 1]
```diff
@@ -13,20 +13,31 @@ class DefinitionParser
...
             ->each(function (Crawler $crawler) {
                 return call_user_func(ParseProperty::fromCrawler($crawler));
-            });
+            }));
 
-        return new TypeCollection(
-            array_filter($types), array_filter($properties)
-        );
+        $constants = [];
+        foreach($types as $type) {
```
Comment (sebastiandedeyne) [inline generator/Parser/DefinitionParser.php]: Add an empty line before `foreach`

File: generator/Parser/DefinitionParser.php [hunk 2]
```diff
@@ -13,20 +13,31 @@ class DefinitionParser
...
+            $constants = array_merge($constants, array_filter($definitions
+                ->query('[typeof="' . $type->resource . '"]')
+                ->each(function (Crawler $crawler) use ($type) {
+                    $constant = call_user_func(ParseConstant::fromCrawler($crawler));
+                    if (!is_null($constant)) {
```
Comment (sebastiandedeyne) [inline generator/Parser/DefinitionParser.php]: Wrap the `if` block with empty lines

File: generator/Parser/DefinitionParser.php [hunk 3]
```diff
@@ -13,20 +13,31 @@ class DefinitionParser
...
+                        $constant->type = $type->name;
+                    }
+                    return $constant;
```
Comment (sebastiandedeyne) [inline generator/Parser/DefinitionParser.php]: Add an empty line before `return`

**Comment by sebastiandedeyne** [comment]
Awesome PR! Requested some minor CS changes :)

**Comment by Gummibeer** [comment]
@sebastiandedeyne have added the requested empty lines

**Comment by sebastiandedeyne** [comment]
Thanks! @josevavia Care to chime in before I merge? :)

**Comment by Gummibeer** [comment]
@sebastiandedeyne now all tests are passed. :)

**Pull request merged**